manufacturers/prob_match_manufacturers.R

```diff
@@ -0,0 +1,186 @@
+# Keying the drug company data
+# S. Kirmer
+# Feb 2017
+# Data4Democracy
+
+library(dplyr)
+library(data.table)
+library(reshape2)
+library(sem)
+library(doBy)
+library(ggplot2)
+library(AER)
+library(ivpack)
+library(readxl)
+library(RecordLinkage)
+library(stringr)
+library(gdata)
+library(stringdist)
+library(stats)
+
+companies_drugs <- read.csv("U://Medicaid_Drug/drugdata_clean.csv", stringsAsFactors = FALSE)
+lobbying <- read.csv("U://Medicaid_Drug/lobbying_data.csv", stringsAsFactors = FALSE)
+#Source for lobbying data: https://data.world/data4democracy/drug-spending/file/Pharma_Lobby.csv
+
+c_names <- unique(companies_drugs$LABELER.NAME)
+l_names <- toupper(unique(lobbying$client))
+
+c_names2 <- cbind(c_names, seq(1:length(c_names))) #Just adding a record indexing identifier
+l_names2 <- cbind(l_names, seq(1:length(l_names)))
+
+# ========== Prep for some probabilistic matching ============ ####
+
+#You want to start with two unique lists, to the best of your ability.
+#In this case we want to cut away things like "INC" or "CO" because that stuff is going to get in the way of clean matches.
+
+c_cut_name <- gsub(" INC| LTD| CORPORATION| CORP| USA| US| COMPANY| & CO| CO| PLC| LLC|\\.|\\,| PHARMACEUTICAL| PHARMACEUTICALS| PHARMA","", c_names2[,"c_names"])
+c_names3 <- as.data.frame(cbind(c_names2, c_cut_name))
+
+l_cut_name <- gsub(" INC| LTD| CORPORATION| CORP| USA| US| COMPANY| & CO| CO| PLC| LLC|\\.|\\,| PHARMACEUTICAL| PHARMACEUTICALS| PHARMA","", l_names2[,"l_names"])
+l_names3 <- as.data.frame(cbind(l_names2, l_cut_name))
+
```

Comment:
Ah, gotcha. I didn't think about it requiring a whole redo! The thing that made me think of it was seeing "ELI LILLY AND COMPANY" somewhere, but you're right, definitely not worth redoing it all.